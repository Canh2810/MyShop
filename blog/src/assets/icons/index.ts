export * from './search'
export * from './sunny'
