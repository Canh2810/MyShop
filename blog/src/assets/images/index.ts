export * from './logo'
export * from './logo-large'
export * from './post-banner.png'
