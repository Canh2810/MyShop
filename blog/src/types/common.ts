export enum TypoVariants {
  TextExtraSmall = 'textExtraSmall',
  TextSmall = 'textSmall',
  TextMedium = 'textMedium',
  TextLarge = 'textLarge',
  TexExtraLarge = 'textExtraLarge',
  HeadingSmall = 'headingSmall',
  HeadingMedium = 'headingMedium',
  HeadingLarge = 'headingLarge',
  HeadingExtraLarge = 'headingExtraLarge',
}

export enum TypoColors {
  Primary = 'textPrimary',
  Secondary = 'textSecondary',
  Tertiary = 'textTertiary',
}

export enum ButtonVariants {
  Container = 'container',
  Outlined = 'outlined',
}

export enum CommonVariants {
  Primary = 'primary',
  Secondary = 'secondary',
}

export enum TextFieldVariants {
  Outlined = 'outlined',
  Filled = 'filled',
}

export enum TextFieldTypes {
  Text = 'text',
  Password = 'password',
}

export enum MenuVariants {
  Vertical = 'vertical',
  Horizontal = 'horizontal',
}

export interface IMenuItem {
  title: string
  href: string
}
