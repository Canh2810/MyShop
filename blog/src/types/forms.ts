export interface ILoginForm {
  email: string
  password: string
}

export type LoginFormField = Exclude<keyof ILoginForm, ''>
