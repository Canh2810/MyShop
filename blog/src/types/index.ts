export * from './common'
export * from './posts'
export * from './forms'
export * from './users'
export * from './response'
