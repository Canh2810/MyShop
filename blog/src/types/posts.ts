import { FavoriteStatus } from '.'

export interface IPost {
  id: string
  category: string
  title: string
  author: {
    avatar: string
    username: string
  }
  date: string
  imageURL: string
  favorite: FavoriteStatus
}
