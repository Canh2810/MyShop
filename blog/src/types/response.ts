import { User, Status } from '.'

export interface LoginResponse {
  status: Status
  message?: string
  data?: User
}
