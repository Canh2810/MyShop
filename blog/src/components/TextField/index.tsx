'use client'

// Libs
import { ChangeEvent, memo } from 'react'

// Types
import { TextFieldTypes, TextFieldVariants } from '@/types'

export interface TextFieldProps {
  variant?: TextFieldVariants
  icon: JSX.Element
  name: string
  placeholder: string
  id: string
  value: string
  type?: TextFieldTypes
  errorMessage?: string
  className?: string
  onChange: (e: ChangeEvent<HTMLInputElement>) => void
}

const TextField = ({
  variant = TextFieldVariants.Outlined,
  icon,
  name,
  placeholder,
  id,
  value,
  type = TextFieldTypes.Text,
  className = '',
  errorMessage,
  onChange,
}: TextFieldProps) => {
  const testFiledStyles = {
    [TextFieldVariants.Outlined]:
      'border border-solid border-light-700 text-dark-50 bg-white dark:bg-dark-200',
    [TextFieldVariants.Filled]: 'bg-light-100 dark:bg-dark-200 text-light-500 ',
  }

  return (
    <div
      className={`relative flex items-center px-4  py-3 rounded-md text-sm ${testFiledStyles[variant]} ${className} `}
    >
      <input
        type={type}
        id={id}
        name={name}
        value={value}
        className="w-full bg-transparent text-light-500 focus:outline-0"
        placeholder={placeholder}
        onChange={onChange}
      />
      {icon}
      {errorMessage && (
        <p className="absolute top-12 text-xs text-danger">{errorMessage}</p>
      )}
    </div>
  )
}

export default memo(TextField)
