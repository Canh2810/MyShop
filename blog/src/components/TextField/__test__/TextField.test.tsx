import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Search } from '@/assets'
import TextField from '..'
import { TextFieldTypes, TextFieldVariants } from '@/types'

describe('TextField component', () => {
  const mockChange = jest.fn()
  const props = {
    icon: <Search />,
    name: 'test',
    placeholder: 'test',
    id: 'test',
    value: '',
    onChange: mockChange,
  }

  it('render with variant is outlined and type is text', () => {
    const { container } = render(<TextField {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('render with variant is filled and type is text', () => {
    const { container } = render(
      <TextField {...props} variant={TextFieldVariants.Filled} />,
    )
    expect(container).toMatchSnapshot()
  })

  it('render with variant is outlined and type is password', () => {
    const { container } = render(
      <TextField {...props} type={TextFieldTypes.Password} />,
    )
    expect(container).toMatchSnapshot()
  })
  it('render with variant is filled and type is password', () => {
    const { container } = render(
      <TextField
        {...props}
        variant={TextFieldVariants.Filled}
        type={TextFieldTypes.Password}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it('should call onChange handler when value changes', async () => {
    const { getByPlaceholderText } = render(<TextField {...props} />)
    const textfield = getByPlaceholderText('test')

    await userEvent.type(textfield, 'abc')

    expect(mockChange).toHaveBeenCalled()
  })
})
