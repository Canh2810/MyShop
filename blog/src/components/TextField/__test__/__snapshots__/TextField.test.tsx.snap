// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList component render with variant is filled and type is password 1`] = `
<div>
  <div
    class="relative flex items-center px-4  py-3 gap-2.5 rounded-md text-sm bg-light-100 dark:bg-dark-200 text-light-500   "
  >
    <input
      class="w-full bg-transparent text-dark-300 focus:outline-0"
      id="test"
      name="test"
      placeholder="test"
      type="password"
      value="test"
    />
    <svg
      fill="none"
      height="16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.91 2a4.909 4.909 0 1 0-.001 9.818 4.909 4.909 0 0 0 0-9.818Z"
        stroke="#52525B"
        stroke-miterlimit="10"
        stroke-width="1.5"
      />
      <path
        d="M10.572 10.572 14 14"
        stroke="#52525B"
        stroke-linecap="round"
        stroke-miterlimit="10"
        stroke-width="1.5"
      />
    </svg>
  </div>
</div>
`;

exports[`PostList component render with variant is filled and type is text 1`] = `
<div>
  <div
    class="relative flex items-center px-4  py-3 gap-2.5 rounded-md text-sm bg-light-100 dark:bg-dark-200 text-light-500   "
  >
    <input
      class="w-full bg-transparent text-dark-300 focus:outline-0"
      id="test"
      name="test"
      placeholder="test"
      type="text"
      value="test"
    />
    <svg
      fill="none"
      height="16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.91 2a4.909 4.909 0 1 0-.001 9.818 4.909 4.909 0 0 0 0-9.818Z"
        stroke="#52525B"
        stroke-miterlimit="10"
        stroke-width="1.5"
      />
      <path
        d="M10.572 10.572 14 14"
        stroke="#52525B"
        stroke-linecap="round"
        stroke-miterlimit="10"
        stroke-width="1.5"
      />
    </svg>
  </div>
</div>
`;

exports[`PostList component render with variant is outlined and type is password 1`] = `
<div>
  <div
    class="relative flex items-center px-4  py-3 gap-2.5 rounded-md text-sm border border-solid border-light-700 text-dark-50 bg-white dark:bg-dark-200  "
  >
    <input
      class="w-full bg-transparent text-dark-300 focus:outline-0"
      id="test"
      name="test"
      placeholder="test"
      type="password"
      value="test"
    />
    <svg
      fill="none"
      height="16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.91 2a4.909 4.909 0 1 0-.001 9.818 4.909 4.909 0 0 0 0-9.818Z"
        stroke="#52525B"
        stroke-miterlimit="10"
        stroke-width="1.5"
      />
      <path
        d="M10.572 10.572 14 14"
        stroke="#52525B"
        stroke-linecap="round"
        stroke-miterlimit="10"
        stroke-width="1.5"
      />
    </svg>
  </div>
</div>
`;

exports[`PostList component render with variant is outlined and type is text 1`] = `
<div>
  <div
    class="relative flex items-center px-4  py-3 gap-2.5 rounded-md text-sm border border-solid border-light-700 text-dark-50 bg-white dark:bg-dark-200  "
  >
    <input
      class="w-full bg-transparent text-dark-300 focus:outline-0"
      id="test"
      name="test"
      placeholder="test"
      type="text"
      value="test"
    />
    <svg
      fill="none"
      height="16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.91 2a4.909 4.909 0 1 0-.001 9.818 4.909 4.909 0 0 0 0-9.818Z"
        stroke="#52525B"
        stroke-miterlimit="10"
        stroke-width="1.5"
      />
      <path
        d="M10.572 10.572 14 14"
        stroke="#52525B"
        stroke-linecap="round"
        stroke-miterlimit="10"
        stroke-width="1.5"
      />
    </svg>
  </div>
</div>
`;
